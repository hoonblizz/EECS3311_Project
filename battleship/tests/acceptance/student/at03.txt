debug_test(easy)
fire([A,5])
fire([E,1])
bomb([B,3],[A,4])
undo
fire([A,1])
undo
undo
redo
bomb([A,1],[A,2])
fire([C,3])
undo
redo
fire([C,3])
bomb([B,3],[B,4])
undo
redo
undo
undo
fire([D,2])
debug_test(medium)
debug_test(hard)
undo
undo
redo

