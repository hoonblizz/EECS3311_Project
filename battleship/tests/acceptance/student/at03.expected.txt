  state 0 OK -> Start a new game
->debug_test(easy)
  state 1 OK -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->fire([A, 5])
  state 2 Invalid coordinate -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->fire([E, 1])
  state 3 Invalid coordinate -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([B, 3],[A, 4])
  state 4 Bomb coordinates must be adjacent -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->undo
  state 5 (= state 3) Invalid coordinate -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->fire([A, 1])
  state 6 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->undo
  state 7 (= state 3) Invalid coordinate -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->undo
  state 8 (= state 2) Invalid coordinate -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->redo
  state 9 (= state 3) Invalid coordinate -> Fire Away!
     1  2  3  4
  A  _  _  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 0/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->bomb([A, 1],[A, 2])
  state 10 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 1/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->fire([C, 3])
  state 11 OK -> 1x1 ship sunk! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->undo
  state 12 (= state 10) OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  h  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 0/8
  Bombs: 1/2
  Score: 0/3 (Total: 0/3)
  Ships: 0/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->h
->redo
  state 13 (= state 11) OK -> 1x1 ship sunk! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->fire([C, 3])
  state 14 Already fired there -> Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->bomb([B, 3],[B, 4])
  state 15 OK -> Hit! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  O  X
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->X;[C, 4]->v
    1x1: [C, 3]->X
->undo
  state 16 (= state 14) Already fired there -> Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->redo
  state 17 (= state 15) OK -> Hit! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  O  X
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 2/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->X;[C, 4]->v
    1x1: [C, 3]->X
->undo
  state 18 (= state 14) Already fired there -> Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->undo
  state 19 (= state 11) OK -> 1x1 ship sunk! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  _  _  _
  Current Game (debug): 1
  Shots: 1/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->fire([D, 2])
  state 20 OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  O  _  _
  Current Game (debug): 1
  Shots: 2/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->debug_test(medium)
  state 21 Game already started -> Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  O  _  _
  Current Game (debug): 1
  Shots: 2/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->debug_test(hard)
  state 22 Game already started -> Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  O  _  _
  Current Game (debug): 1
  Shots: 2/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->undo
  state 23 (= state 21) Game already started -> Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  O  _  _
  Current Game (debug): 1
  Shots: 2/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->undo
  state 24 (= state 20) OK -> Miss! Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  O  _  _
  Current Game (debug): 1
  Shots: 2/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
->redo
  state 25 (= state 21) Game already started -> Keep Firing!
     1  2  3  4
  A  O  O  _  _
  B  _  _  _  v
  C  _  _  X  v
  D  _  O  _  _
  Current Game (debug): 1
  Shots: 2/8
  Bombs: 1/2
  Score: 1/3 (Total: 1/3)
  Ships: 1/2
    2x1: [B, 4]->v;[C, 4]->v
    1x1: [C, 3]->X
